Common/Securities/Volatility/StandardDeviationOfReturnsVolatilityModel.cs

```diff
@@ -0,0 +1,122 @@
+*/
+
+using System;
+using MathNet.Numerics.Statistics;
+using QuantConnect.Data;
+using QuantConnect.Indicators;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Securities
+{
+    /// <summary>
+    /// Provides an implementation of <see cref="IVolatilityModel"/> that computes the
+    /// annualized sample standard deviation of daily returns as the volatility of the security
+    /// </summary>
+    public class StandardDeviationOfReturnsVolatilityModel : IVolatilityModel
+    {
+        private bool _needsUpdate;
+        private decimal _volatility;
+        private DateTime _lastUpdate = DateTime.MinValue;
+        private decimal _lastPrice;
+        private readonly TimeSpan _periodSpan = TimeSpan.FromDays(1);
+        private readonly object _sync = new object();
+        private readonly RollingWindow<double> _window;
+
+        /// <summary>
+        /// Gets the volatility of the security as a percentage
+        /// </summary>
+        public decimal Volatility
+        {
+            get
+            {
+                lock (_sync)
+                {
+                    if (_window.Count < 2)
+                    {
+                        return 0m;
+                    }
+
+                    if (_needsUpdate)
+                    {
+                        _needsUpdate = false;
+                        var std = _window.StandardDeviation().SafeDecimalCast();
+                        _volatility = std * (decimal)Math.Sqrt(252.0);
+                    }
+                }
+                return _volatility;
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StandardDeviationOfReturnsVolatilityModel"/> class
+        /// </summary>
+        /// <param name="periods">The number of periods (days) to wait until updating the value</param>
+        public StandardDeviationOfReturnsVolatilityModel(int periods)
+        {
+            if (periods < 2) throw new ArgumentOutOfRangeException("periods", "'periods' must be greater than or equal to 2.");
+            _window = new RollingWindow<double>(periods);
+        }
+
+        /// <summary>
+        /// Updates this model using the new price information in
+        /// the specified security instance
+        /// </summary>
+        /// <param name="security">The security to calculate volatility for</param>
+        /// <param name="data"></param>
+        public void Update(Security security, BaseData data)
+        {
+            var timeSinceLastUpdate = data.EndTime - _lastUpdate;
+            if (timeSinceLastUpdate >= _periodSpan)
+            {
+                lock (_sync)
+                {
+                    if (_lastPrice > 0.0m)
+                    {
+                        _needsUpdate = true;
+                        _window.Add((double)(data.Price / _lastPrice) - 1.0);
+                    }
+                }
+                _lastUpdate = data.EndTime;
+                _lastPrice = data.Price;
```

Comment:
Fixed this in both models. NB we have to remember to adjust those later when we introduce zero/negative instrument prices (futures spreads, TAS, etc).